src/hooks/useCoinListEditOptions.ts

```diff
@@ -0,0 +1,141 @@
+import { difference } from 'lodash';
+import { useCallback, useMemo, useRef } from 'react';
+import { useMMKVObject } from 'react-native-mmkv';
+import { useDispatch } from 'react-redux';
+import { atom, useRecoilState, useSetRecoilState } from 'recoil';
+import useAccountSettings from './useAccountSettings';
+import actions from '@rainbow-me/helpers/editOptionTypes';
+import { setHiddenCoins as reduxSetHiddenCoins } from '@rainbow-me/redux/editOptions';
+
+const selectedItemsAtom = atom<string[]>({
+  default: [],
+  key: 'selectedItemsAtom',
+});
+
+export default function useCoinListEditOptions() {
+  const { accountAddress: address } = useAccountSettings();
+
+  const setSelectedItems = useSetRecoilState(selectedItemsAtom);
+  const [hiddenCoins = []] = useMMKVObject<string[]>('hidden-coins-' + address);
+
+  const [pinnedCoins = []] = useMMKVObject<string[]>('pinned-coins-' + address);
+  const pushSelectedCoin = useCallback(
+    (item: string) =>
+      setSelectedItems(prev => {
+        return [...prev.filter(i => i !== item), item];
+      }),
+    [setSelectedItems]
+  );
+
+  const removeSelectedCoin = useCallback(
+    (item: string) =>
+      setSelectedItems(prev => [...prev.filter(i => i !== item)]),
+    [setSelectedItems]
+  );
+
+  const toggleSelectedCoin = useCallback(
+    (item: string) =>
+      setSelectedItems(prev => {
+        if (prev.includes(item)) {
+          return [...prev.filter(i => i !== item)];
```

Comment:
Filter returns a new array anyway, we don't need to do another copy
```suggestion
          return prev.filter(i => i !== item);
```

Reply:
I would change it in other places too